### PR TITLE
[3.3.5][6.x]Core/Unit: Do not remove auras from Pets and Guardians after evading

### LINKED ISSUE #14347: [3.3.5][6.x] Guardians evade issue
rev. 68c4ea314443

After a combat or in_combat Guardians will loose all auras (harmful or not) this breaks many quests, for pets this issue doesn't happen.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9480287-guardians-evade-issue?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Lumber-Jack-2**
- this adds free space for further adjustments
- this fixes #14347 
- originally work by @mik1893

### CODE REVIEW

**Comment by Rushor** [comment]
this also fixes following bug:
1) start combat as a rogue with a dk with active ghoul
2) ghoul starts attacking rogue
3) rogue casts blind on ghoul
4) rogue casts vanish on hisself

result: ghoul will evade and blind will be removed. blind should never be removed. this pr fixes that issue

File: src/server/game/AI/CreatureAI.cpp [hunk 1]
```diff
@@ -259,7 +259,7 @@ bool CreatureAI::_EnterEvadeMode(EvadeReason /*why*/)
 
     // don't remove vehicle auras, passengers aren't supposed to drop off the vehicle
     // don't remove clone caster on evade (to be verified)
```
Comment (offl) [inline src/server/game/AI/CreatureAI.cpp]: Now these comments possible should be moved to [Unit.cpp](https://github.com/TrinityCore/TrinityCore/pull/16503/files#diff-4522eeffcbc961602e86620011a2d579R4121)

Comment (Rushor) [inline src/server/game/AI/CreatureAI.cpp]: done

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -4113,6 +4113,16 @@ void Unit::RemoveArenaAuras()
     }
 }
 
+void Unit::RemoveAurasOnEvade()
+{
+    if (IsPet() || IsGuardian()) // if pet or guardian shouldn't lose any aura
```
Comment (Shauren) [inline src/server/game/Entities/Unit/Unit.cpp]: I wonder if this should maybe extend to all player owned creatures, not just hunter/warlock pets and guardians

Comment (Rushor) [inline src/server/game/Entities/Unit/Unit.cpp]: do we have a check for such creatures? (personally i never saw more than IsPet() and IsGuardian() in any check for player owned creatures)

Comment (ccrs) [inline src/server/game/Entities/Unit/Unit.cpp]: IsCharmedOwnedByPlayerOrPlayer

Comment (Rushor) [inline src/server/game/Entities/Unit/Unit.cpp]: ah thx, update soon

Comment (Rushor) [inline src/server/game/Entities/Unit/Unit.cpp]: done

File: src/server/game/Entities/Unit/Unit.cpp [hunk 3]
```diff
@@ -4113,6 +4113,16 @@ void Unit::RemoveArenaAuras()
     }
 }
 
+void Unit::RemoveAurasOnEvade()
+{
+    if (IsCharmedOwnedByPlayerOrPlayer()) // if it is a player owned creature it should not remove the aura
```
Comment (Rushor) [inline src/server/game/Entities/Unit/Unit.cpp]: any news on this? would you like to improve sth. here?

Comment (tkrokli) [inline src/server/game/Entities/Unit/Unit.cpp]: Looks good to me™. I am fairly sure @ccrs meant it should look like this.

**Comment by jackpoz** [comment]
@Rushor is this PR ready to be merged ?

**Comment by Rushor** [comment]
yep, i tested this since 2 months now - i didn't found any new issues with it

**Comment by jackpoz** [comment]
merge it then please

**Pull request merged**